### PR TITLE
Add env variable for timezone where the makerspace is

### PR DESCRIPTION
**Pull request by BerglundDaniel**
Solves #467 

## Summary by CodeRabbit

- **New Features**
	- Introduced a method to retrieve the local timezone for the makerspace.
	- Added a command-line argument for interactive timezone input during environment setup.
	- Updated accounting file download functionality to utilize the new timezone method.

- **Bug Fixes**
	- Corrected indentation in the control flow of an existing method.
	- Improved error handling for missing MySQL password configuration.

- **Chores**
	- Modified the Docker configuration to include timezone settings for the API service.

### CODE REVIEW

File: api/src/service/config.py [hunk 1]
```diff
@@ -91,5 +92,14 @@ def get_admin_url(path):
     return f"{host}{path}"
 
 
+def get_makerspace_local_timezone() -> ZoneInfo:
+    zone_str = config.get("MAKERSPACE_LOCAL_TIMEZONE")
+    try:
+        zone = ZoneInfo(zone_str)
+    except Exception as e:
+        raise (f"Variable MAKERSPACE_LOCAL_TIMEZONE not set correctly in .env, failed due to {e}")
```
Comment (emanuelen5) [inline api/src/service/config.py:100]: You should raise an exception object, not a string

**Review by emanuelen5 — CHANGES_REQUESTED** [review]

File: create_env.py [hunk 2]
```diff
@@ -38,6 +38,7 @@
     "FIRSTRUN_AUTO_ADMIN_PASSWORD": "",
     "ELKS46_API_USER": "",
     "ELKS46_API_KEY": "",
+    "MAKERSPACE_LOCAL_TIMEZONE": "Europe/Stockholm",
```
Comment (HalfVoxel) [inline create_env.py:41]: This should be a prompt with a good default. It should link to where one can find all valid timezones, and it should validate that the timezone the user sets is a valid timezone.

**Review by HalfVoxel — CHANGES_REQUESTED** [review]

File: create_env.py [hunk 3]
```diff
@@ -2,10 +2,31 @@
...
 parser = argparse.ArgumentParser(description="Create a default '.env' file with secrets if it doesn't exist")
 parser.add_argument("--force", "-f", dest="force", action="store_true", help="overwrite existing '.env' file")
+parser.add_argument("--interactive", "-i", dest="interactive", action="store_true", help="run with interactive choices")
+
 args = parser.parse_args()
 
+if not args.force and os.path.isfile(".env"):
+    print(".env file already exists, touching")
+    os.utime(".env", None)
+    exit()
```
Comment (emanuelen5) [inline create_env.py:16]: Add the exit code to make it explicit, i.e. `exit(0)`

File: create_env.py [hunk 4]
```diff
@@ -2,10 +2,31 @@
...
+
+if args.interactive:
+    while True:
+        zone_str = input("What is the makerspace timezone? [Y/n]: ")
+
+        if zone_str in available_timezones():
+            makerspace_local_timezone = zone_str
+            break
+        else:
+            print(f"Timezone {zone_str} is not in the list of available zones")
```
[the remark this answers is not in the document]
Comment (emanuelen5) [reply create_env.py:26]: ^^^ Good bot

[the remark this answers is not in the document]
Comment (emanuelen5) [reply create_env.py:26]: @BerglundDaniel I think you should add something similar to what the bot suggested here:


```diff
 while True:
-    zone_str = input("What is the makerspace timezone? [Y/n]: ")
+    zone_str = input("Enter the makerspace timezone (e.g., 'Europe/Stockholm'): ")

     if zone_str in available_timezones():
         makerspace_local_timezone = zone_str
         break
     else:
-        print(f"Timezone {zone_str} is not in the list of available zones")
+        print(f"Timezone '{zone_str}' is not valid.")
+        print("Please enter a valid timezone (e.g., 'Europe/Stockholm').")
+        print("Refer to https://en.wikipedia.org/wiki/List_of_tz_database_time_zones for a list of valid timezones.")
```

File: create_env.py [hunk 5]
```diff
@@ -2,10 +2,31 @@
...
 args = parser.parse_args()
 
+if not args.force and os.path.isfile(".env"):
+    print(".env file already exists, touching")
+    os.utime(".env", None)
+    exit(0)
+
+if args.interactive:
+    while True:
+        zone_str = input("What is the makerspace timezone? [Y/n]: ")
```
Comment (emanuelen5) [inline create_env.py:20]: The input should not be "Y" or "n" as suggested by the message
```diff
- zone_str = input("What is the makerspace timezone? [Y/n]: ")
+ zone_str = input("What is the makerspace timezone?: ")
```

**Review by emanuelen5 — APPROVED** [review]

**Pull request merged**